packages/flutter/lib/src/widgets/router.dart

```diff
@@ -42,15 +42,51 @@ class RouteInformation {
   /// Creates a route information object.
   ///
   /// The arguments may be null.
-  const RouteInformation({this.location, this.state});
+  const RouteInformation({
+    @Deprecated(
+      'Passes Uri.parse(location) to uri parameter instead. '
+      'MaterialApp never introduces its own MediaQuery; the View widget takes care of that. '
+      'This feature was deprecated after v3.7.0-29.0.pre.'
+    )
+    String? location,
+    Uri? uri,
+    this.state,
+  }) : _location = location,
+       _uri = uri,
+       assert((location != null) != (uri != null));
 
   /// The location of the application.
   ///
   /// The string is usually in the format of multiple string identifiers with
   /// slashes in between. ex: `/`, `/path`, `/path/to/the/app`.
   ///
   /// It is equivalent to the URL in a web application.
-  final String? location;
+  String get location {
```

Comment:
Should this getter be deprecated as well? Or do you plan to keep it around even after the constructor arg is removed?